decoder.go

```diff
@@ -258,6 +227,52 @@ func DecodeTime(bz []byte) (t time.Time, n int, err error) {
 	return
 }
 
+func decodeSeconds(bz *[]byte) (int64, int, error) {
+	// Optionally decode field number 1 and Typ3 (8Byte).
+	// only slide if we need to:
+	n := 0
+	fieldNum, typ, _n, err := decodeFieldNumberAndTyp3(*bz)
+	if err != nil {
+		return 0, n, err
+	}
+	if fieldNum == 1 && typ == Typ3_8Byte {
+		slide(bz, &n, _n)
+		_n = 0
+		sec, _n, err := DecodeInt64(*bz)
+		if slide(bz, &n, _n) && err != nil {
+			return 0, n, err
+		} else {
+			return sec, n, err
+		}
+	} else if fieldNum == 2 && typ == Typ3_4Byte {
+		// skip: do not slide, no error, will read again
+		return 0, n, nil
+	} else {
+		return 0, n, fmt.Errorf("expected field number 1 <8Bytes> or field number 2 <4Bytes> , got %v", fieldNum)
+	}
+}
+
+func decodeNanos(bz *[]byte, n *int) (int32, error) {
+	// Optionally decode field number 2 and Typ3 (4Byte).
+	//var fieldNum, typ, _n = uint32(0), Typ3(0x00), int(0)
```

Comment:
Maybe wec an rewrite this comment to show the layout in the byte slice, as an example check [this section](https://github.com/tendermint/go-amino#prefix-bytes-to-identify-the-concrete-type) from the README.

Reply:
Thanks! I just forgot to delete this comment. I don't think there should be a general comment about how the byteslice will look here. It's very general (typ3AndFieldnum | actualBytesOfField) and not special. I'll delete the comment. A more public facing documentation should explain how time is encoded though.